Modify rule S956: LaYC format (#2512)
SonarSource/rspec: rules/S956/cfamily/rule.adoc

== Why is this an issue?

The `#include` directive is a preprocessor directive that tells the compiler to insert the contents of a file in the source code.

However, the standard only allows the `#include` directive to be followed by angle brackets (`<filename.h>`) or double quotes (`"filename.h"`).

> If the `#include` contains macro names, the replacement list of preprocessing tokens must also follow this rule:

Reviewer: ```suggestion
If the `#include` directive contains macro names, then the result of expanding them must also follow this rule:
```
I didn't know about "replacement-lists" (which seems to be the term used by the standard to describe the `#define` directive grammar). Maybe macro expansion is more common here? WDYT?

Author: I totally agree, thanks!
